sdk/core/Azure.Core/src/ProcessingContext.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Contains information related to the processing of the <see cref="HttpMessage"/>.
+    /// </summary>
+    public class ProcessingContext
+    {
+        /// <summary>
+        /// The start time of the operation.
+        /// </summary>
+        public DateTimeOffset OperationStartTime
+        {
+            get
+            {
+                if (_message != null)
+                {
+                    return _message.OperationStartTime;
+                }
+
+                return _operationStartTime;
```

Comment:
Why do we have these two modes: one when message is null and one when it's not? i.e. why would we allo this type to not be associated with a message?